src/overlays/actors/ovl_En_Look_Nuts/z_en_look_nuts.c

```diff
@@ -71,34 +92,292 @@ static DamageTable D_80A68630 = {
+    Animation_Change(&this->skelAnime, &gDekuPalaceGuardWalkAnim, 1.0f, 0.0f,
+                     Animation_GetLastFrame(&gDekuPalaceGuardWalkAnim), 2, -10.0f);
+    this->waitTimer = Rand_S16Offset(1, 3);
+    this->unk238.y = 10000.0f;
+
+    if (Rand_ZeroOne() < 0.5f) {
+        this->unk238.y = -10000.0f;
+    }
+    this->unk21A = 10;
+    this->state = WAITING_STATE;
+    this->actionFunc = EnLookNuts_StandAndWait;
+}
+
+// Patrol Guards will stand in place and wait for a maximum of ~12 frames.
+void EnLookNuts_StandAndWait(EnLookNuts* this, GlobalContext* globalCtx) {
+    s16 randOffset;
 
-extern ColliderCylinderInit D_80A68600;
-extern DamageTable D_80A68630;
+    SkelAnime_Update(&this->skelAnime);
+    Math_ApproachZeroF(&this->actor.speedXZ, 0.3f, 1.0f);
+    if ((func_801690CC(globalCtx) == 0) && (D_80A6862C == 0) && (this->unk21A == 0)) {
+        this->unk21A = 10;
+        switch (this->waitTimer) {
+            case 0:
+            case 1:
+            case 2:
+            case 3:
+            case 4:
+                this->waitTimer++;
+                this->unk238.y *= -1.0f;
+                break;
+            case 5:
+                this->unk238.y = 0.0f;
+                randOffset = Rand_S16Offset(1, 2);
+                this->unk21A = 0;
+                this->waitTimer += randOffset;
+                break;
+            case 6:
+                if (fabsf(this->unk238.y - this->headRotation.y) < 10.0f) {
+                    this->waitTimer = 10;
+                    this->unk238.x = 4000.0f;
+                    this->unk21A = 5;
+                }
+                break;
+            case 7:
+                if (fabsf(this->unk238.y - this->headRotation.y) < 10.0f) {
+                    this->unk238.z = 4000.0f;
+                    this->waitTimer++;
+                }
+                break;
+            case 8:
+                this->waitTimer = 10;
+                this->unk21A = 20;
+                this->unk238.z = -8000.0f;
+                break;
+            case 10:
+                Math_Vec3f_Copy(&this->unk238, &gZeroVec3f);
+                this->waitTimer = 11;
+                break;
+            case 11:
+                if ((fabsf(this->headRotation.x) < 30.0f) && (fabsf(this->headRotation.y) < 30.0f) &&
+                    (fabsf(this->headRotation.z) < 30.0f)) {
+                    this->waitTimer = 12;
+                }
+                break;
+        }
+        if (this->waitTimer == 12) {
+            this->waitTimer = 0;
+            EnLookNuts_SetupPatrol(this);
+        }
+    }
+}
 
-extern UNK_TYPE D_06000430;
-extern UNK_TYPE D_06002B6C;
+void EnLookNuts_DetectedPlayer(EnLookNuts* this, GlobalContext* globalCtx) {
+    Animation_Change(&this->skelAnime, &gDekuPalaceGuardWalkAnim, 2.0f, 0.0f,
+                     Animation_GetLastFrame(&gDekuPalaceGuardWalkAnim), 0, -10.0f);
+    this->state = RUNNING_TO_PLAYER_STATE;
+    this->unk21A = 0x12C;
```

Comment:
```suggestion
    this->unk21A = 300;
```
Looks like this is another timer, too... is it just to do with when to proceed to the waiting thing?